python/arcticdb/arctic.py

```diff
@@ -103,6 +103,8 @@ def __init__(self, uri: str, encoding_version: EncodingVersion = DEFAULT_ENCODIN
                 |                           | "/etc/pki/tls/cacert.pem"                             OpenELEC                                                                                                |
                 |                           | "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"   CentOS/RHEL 7                                                                                           |
                 |                           | "/etc/ssl/cert.pem"                                   Alpine Linux                                                                                            |
+                |                           | WARNING for WINDOWS USER: Not leaving this empty will switch the backend support of Azure SDK from winhttp to libcurl. If ca cert path is needed to be        |
```

Comment:
This warning is a bit ungrammatical, and this docstring is getting massive!

But it makes me think, is it ever correct for a Windows user to set `CA_cert_path`? Shouldn't we leave it to Windows to determine (it doesn't have the Linux "many locations" problem). Should we throw if a Windows user sets this to a non-empty value (and document that)?

I suggest throw if non-empty rather than throw if present so that users can still use the same code across Linux/Windows, just with a bit of config for the cert path location that differs by OS.

Reply:
One more, docstring will be massively reduced once the ca path and directory can be automatically assigned. The commit is coming soon.

Reply:
OK, fair enough. Can I suggest we log a warning on Windows when this is set, then?

What is your comment regarding 

`using libcurl as backend shows buggy behaviour on ssl ca cert verification.` about? What was the buggy behaviour?

It would be good to explain the impact of the change of backend a bit in the log warning.

Reply:
Text suggestion for this docstring if we keep it:

```
WARNING for Windows users: Setting this to anything besides an empty string will switch the backend support of Azure SDK from winhttp to libcurl. <EXPLANATION OF IMPACT>

If CA Cert path needs to be specified, set it in the Windows settings rather than using the option, so that winhttp will be used.
```

Reply:
Could it be that Azure switches to libcurl backend here because it assumes that if `CA_Cert_Path` is set then it can't be on Windows, or something? It would be good to see if there is anything in the Azure git blame to understand why this happens at all.

Reply:
> OK, fair enough. Can I suggest we log a warning on Windows when this is set, then?
> 
> What is your comment regarding
> 
> `using libcurl as backend shows buggy behaviour on ssl ca cert verification.` about? What was the buggy behaviour?
> 
> It would be good to explain the impact of the change of backend a bit in the log warning.

Verification fails if SAS Token is used with libcurl on Windows. The problem is gone once switched to winhttp.
I use the word `buggy` as the fail is weird, e.g. `list_libraries()` is fine but `read` gives error.
And probably it will be good idea to give user a warning regarding it.

Reply:
After discussion with @poodlewars, exception will be thrown if `CA_cert_path` or `CA_cert_dir` is being set. 
Further compacting of the docstring is handled in https://github.com/man-group/ArcticDB/pull/867